report/src/report.cpp

```diff
@@ -510,24 +510,48 @@ auto scp::get_log_verbosity(char const* str) -> sc_core::sc_verbosity {
     if (it != lut.end())
         return it->second;
     if (sc_core::sc_get_current_object()) {
-        auto param_name = std::string(str) + "." SCP_LOG_LEVEL_PARAM_NAME;
-        auto h = cci::cci_get_broker().get_param_handle<unsigned>(param_name);
-        if (h.is_valid()) {
-            sc_core::sc_verbosity ret = verbosity.at(
-                std::min<unsigned>(h.get_value(), verbosity.size() - 1));
-            lut[k] = ret;
-            return ret;
-        } else {
-            auto val = cci::cci_get_broker().get_preset_cci_value(param_name);
-            auto global_verb = static_cast<sc_core::sc_verbosity>(
-                ::sc_core::sc_report_handler::get_verbosity_level());
-            sc_core::sc_verbosity ret = val.is_int()
-                                            ? verbosity.at(std::min<unsigned>(
-                                                  val.get_int(),
-                                                  verbosity.size() - 1))
-                                            : global_verb;
-            lut[k] = ret;
-            return ret;
+        string current_name = std::string(str);
+        while (true) {
+            string param_name = (current_name.empty())
+                                    ? SCP_LOG_LEVEL_PARAM_NAME
+                                    : current_name +
+                                          "." SCP_LOG_LEVEL_PARAM_NAME;
+
+            auto h = cci::cci_get_broker().get_param_handle(param_name);
```

Comment:
using this directly prevents the use of the SCP* macros from sc_main. You will get
`Error: /Accellera/CCI/cci_get_broker: You may not request a broker from outside the SystemC hierarchy (use cci::cci_get_global_broker)` at runtime.

It might be better to use
```
auto broker = sc_core::sc_get_current_object()? cci::cci_get_broker(): cci::cci_get_global_broker(originator);
auto h = broker.get_param_handle(param_name);
```

and adding 
```
#ifdef HAS_CCI
thread_local cci::cci_originator originator("reporting");
#endif
```` 
in the anonymous namespace.

Reply:
Good catch and NICE idea !

Reply:
Well, I'm already bitten by that....
It should be changed in line 528 as well